src/frontend/screens/ObservationFields/Question.tsx

```diff
@@ -22,5 +23,9 @@ export const Question = ({field}: QuestionProps) => {
     return <SelectMultiple field={field as SelectMultipleField} />;
   }
 
+  if (field.type === 'number') {
```

Comment:
Could this be a map to be a little more scalable/ elegant? Something like 
```
const FIELD_COMPONENTS: Record<string, React.ComponentType<{ field: Field }>> = {
  selectOne: SelectOne,
  selectMultiple: SelectMultiple,
  text: TextArea,
  number: Number,
};

export const Question = ({ field }: QuestionProps) => {
  const FieldComponent = FIELD_COMPONENTS[field.type] || TextArea;
  return <FieldComponent field={field} />;
};

```

Reply:
I tried this but there are some annoying typing issues that arise.  the `field` prop for each of these components are actually slightly modified subsets of `Field` in schema. This is because the React Component property has certain properties that are necessary for the UI but not necessary for the actual schema.

I tried widening the definition of `React.ComponentType<{ field: Field }>>` to include the other types of props, but that lead into a TS battle of narrowing props that ended up making the mapping still complicated